OrbitGl/Batcher.cpp

```diff
@@ -155,14 +175,41 @@ const TextBox* Batcher::GetTextBox(PickingId id) {
   return nullptr;
 }
 
-void Batcher::GetBoxGradientColors(const Color& color, std::array<Color, 4>* colors) {
+void Batcher::GetBoxGradientColors(const Color& color, std::array<Color, 4>* colors,
+                                   ShadingDirection shading_direction) {
   const float kGradientCoeff = 0.94f;
   Vec3 dark = Vec3(color[0], color[1], color[2]) * kGradientCoeff;
-  (*colors)[0] = Color(static_cast<uint8_t>(dark[0]), static_cast<uint8_t>(dark[1]),
-                       static_cast<uint8_t>(dark[2]), color[3]);
-  (*colors)[1] = (*colors)[0];
-  (*colors)[2] = color;
-  (*colors)[3] = color;
+
+  switch (shading_direction) {
+    case ShadingDirection::kLeftToRight:
+      (*colors)[0] = Color(static_cast<uint8_t>(dark[0]), static_cast<uint8_t>(dark[1]),
+                           static_cast<uint8_t>(dark[2]), color[3]);
+      (*colors)[1] = (*colors)[0];
+      (*colors)[2] = color;
+      (*colors)[3] = color;
+      break;
+    case ShadingDirection::kRightToLeft:
+      (*colors)[2] = Color(static_cast<uint8_t>(dark[0]), static_cast<uint8_t>(dark[1]),
+                           static_cast<uint8_t>(dark[2]), color[3]);
+      (*colors)[3] = (*colors)[2];
+      (*colors)[0] = color;
+      (*colors)[1] = color;
+      break;
+    case ShadingDirection::kTopToBottom:
+      (*colors)[0] = Color(static_cast<uint8_t>(dark[0]), static_cast<uint8_t>(dark[1]),
+                           static_cast<uint8_t>(dark[2]), color[3]);
+      (*colors)[3] = (*colors)[0];
+      (*colors)[1] = color;
+      (*colors)[2] = color;
+      break;
+    case ShadingDirection::kBottomToTop:
+      (*colors)[1] = Color(static_cast<uint8_t>(dark[0]), static_cast<uint8_t>(dark[1]),
+                           static_cast<uint8_t>(dark[2]), color[3]);
+      (*colors)[2] = (*colors)[1];
+      (*colors)[0] = color;
+      (*colors)[3] = color;
+      break;
+  }
```

Comment:
Could "Color(static_cast<uint8_t>(dark[0]), static_cast<uint8_t>(dark[1]),  static_cast<uint8_t>(dark[2]), color[3]);" be merged into a single "dark_color" temporary variable? I think it would be more obvious that 2 vertices are dark and the two others are the original color. WDYT?

could

Reply:
Good idea, that makes the code look a lot cleaner, thanks!